GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -21,8 +24,16 @@ public abstract class HttpRequestor : IDisposable
 
         private readonly ProductInfoHeaderValue userAgentHeader;
 
+        private readonly GitAuthentication authentication;
+
+        private readonly Lazy<X509Store> store = new Lazy<X509Store>(() =>
+        {
+            X509Store s = new X509Store();
+            s.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
+            return s;
+        });

```

Comment:
Can we move most of this code closer to where it belongs?  For example this `store` should be in the `GitSslSettings` class along with the `LoadCertificatePassword` and `LoadCertificate`.  Probably want to rename the class to `GitSsl` since it would be more than just the settings.

Then I would have a method on the `GitAuthentication` class that is 
```
void SetupSslIfNeeded(ITracer tracer, HttpClientHandler httpClientHandler, GitProcess gitProcess)
```
Which does all the work in the constructor of this class so it would only have 
```
HttpClientHandler httpClientHandler = new HttpClientHandler() { UseDefaultCredentials = true };
enlistment.Authentication.SetupSslIfNeeded(tracer, httpClientHandler, enlistment.CreateGitProcess());
```

Reply:
Great suggestions, will do!